components/brave_wallet/browser/brave_wallet_utils.cc

```diff
@@ -1414,14 +1428,42 @@ void RemoveHiddenNetwork(PrefService* prefs,
   });
 }
 
-std::string GetCurrentChainId(PrefService* prefs, mojom::CoinType coin) {
-  const auto& selected_networks = prefs->GetDict(kBraveWalletSelectedNetworks);
-  const std::string* chain_id =
-      selected_networks.FindString(GetPrefKeyForCoinType(coin));
-  if (!chain_id)
-    return std::string();
+std::string GetCurrentChainId(PrefService* prefs,
+                              mojom::CoinType coin,
+                              const absl::optional<url::Origin>& origin) {
+  if (!origin) {
+    return GetCurrentChainId(prefs, coin);
+  }
+  const auto& selected_networks =
+      prefs->GetDict(kBraveWalletSelectedNetworksPerOrigin);
+  const auto* coin_dict =
+      selected_networks.FindDict(GetPrefKeyForCoinType(coin));
+  if (!coin_dict) {
+    return GetCurrentChainId(prefs, coin);
+  }
+  const auto* chain_id_str = coin_dict->FindString(origin->Serialize());
+  if (!chain_id_str) {
+    return GetCurrentChainId(prefs, coin);
+  }
 
-  return base::ToLowerASCII(*chain_id);
+  return base::ToLowerASCII(*chain_id_str);
+}
+
+void SetCurrentChainId(PrefService* prefs,
+                       mojom::CoinType coin,
+                       const absl::optional<url::Origin>& origin,
+                       const std::string& chain_id) {
+  if (!origin) {
+    ScopedDictPrefUpdate update(prefs, kBraveWalletSelectedNetworks);
+    update->Set(GetPrefKeyForCoinType(coin), chain_id);
+  } else {
+    if (origin->opaque() || origin->scheme() == url::kFileScheme) {
+      return;
+    }
```

Comment:
I think it's better to return false to callers in this case.

Reply:
fixed in f5a491afdc05ed6049217a61fb9823392968f7a2